From eb16cd144ce8b01d029e3425577a9156a7b19674 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Mon, 12 Jan 2026 17:14:38 +0100
Subject: [PATCH 949/954] feat(efi): add specific GUIDs for metadata partitions
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

metadata1 and metadata2 must be differentiated even though they share
the same partition type GUID. Add two fixed partition GUIDs to uniquely
identify each metadata partition.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 3d36d41e6296b605840449c812591c8103ae886c feat(efi): add specific GUIDs for metadata partitions]
Change-Id: Ie490651c65a593d401cd7055520018421f38ee9d
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/523844
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 include/drivers/partition/efi.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/drivers/partition/efi.h b/include/drivers/partition/efi.h
index 59dea57017..6d5466f31b 100644
--- a/include/drivers/partition/efi.h
+++ b/include/drivers/partition/efi.h
@@ -35,6 +35,14 @@ static inline void *guidcpy(void *dst, const void *src)
 	EFI_GUID(0x8A7A84A0U, 0x8387U, 0x40F6U, \
 		 0xABU, 0x41U, 0xA8U, 0xB9U, 0xA5U, 0xA6U, 0x0DU, 0x23U)
 
+#define FWU_METADATA1_PART_GUID \
+	EFI_GUID(0x5F91C128U, 0xE120U, 0x48D7U, \
+		 0xBBU, 0x64U, 0x9FU, 0xA9U, 0xAEU, 0xD4U, 0xC7U, 0xE3U)
+
+#define FWU_METADATA2_PART_GUID \
+	EFI_GUID(0xF469F981U, 0x5985U, 0x4206U, \
+		 0x8FU, 0xB3U, 0x83U, 0x99U, 0x56U, 0x03U, 0x5AU, 0x65U)
+
 #define NULL_GUID \
 	EFI_GUID(0x00000000U, 0x0000U, 0x0000U, 0x00U, 0x00U, \
 		 0x00U, 0x00U, 0x00U, 0x00U, 0x00U, 0x00U)
-- 
2.53.0

